src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.VSCodeWorkspaces/VSCodeHelper/VSCodeInstances.cs

```diff
@@ -122,7 +122,9 @@ public static void LoadVSCodeInstances()
 
                             if (version != string.Empty)
                             {
-                                instance.AppData = Path.Combine(_userAppDataPath, version);
+                                var portableData = Path.Join(iconPath, "data");
+                                instance.AppData = Directory.Exists(portableData) ? Path.Join(portableData, "user-data") : Path.Combine(_userAppDataPath, version);
```

Comment:
This will ignore all the versions installed on the computer. So If we have Code, Code-Insiders and/or VSCodium all will have  by default the path of the portable vscode version. So my recomendation is validation  before the foreach. if a folder exists in the enviroment variables with that structure, add it as a path. And then if the folder is of type portable put the AppData of the portable version.
![image](https://user-images.githubusercontent.com/20097801/179366937-204c9abf-d745-43ce-a5fa-79b929067ffe.png)
@bkmeneguello

Reply:
Actually this doesn't break functionality of installed instances as there is no 'data' dir in their iconPath. However, this requires having `vscode_portable_dir/bin` in PATH env var to work. I think we should at least note this somewhere

Reply:
You are right. For me this is ok then.

Reply:
Lets put the PATH warning in the subheader text and docs as well.

Reply:
The warning here is that "Portable instances of of VSCode need to in the PATH environment variable" work, right @stefansjfw ?

Reply:
Correct. Because VS Code plugin goes through PATH env var and looks for executable and related data